opm/common/util/numeric/cmp.hpp

```diff
@@ -0,0 +1,43 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef COMMON_UTIL_NUMERIC_CMP
+#define COMMON_UTIL_NUMERIC_CMP
+
+#include <cstddef>
+#include <vector>
+
+namespace Opm {
+    namespace cmp {
+        const double abs_epsilon = 1e-8;
+        const double rel_epsilon = 1e-5;
+        
+
+        bool double_equal(double value1 , double value2 , double abs_eps , double rel_eps);
+        bool double_equal(double value1 , double value2);
+
+        bool double_vector_equal(const std::vector<double>& v1,const std::vector<double>& v2, double abs_eps , double rel_eps);
+        bool double_vector_equal(const std::vector<double>& v1,const std::vector<double>& v2);
+
+        int double_memcmp(const double * p1 , const double *p2 , size_t num_elements, double abs_eps , double rel_eps);
```

Comment:
I think double_array_equal() would be a good name, but if you prefer to keep the C link
it's ok with me (I did get confused upon first reading though, mixing memcmp and memcpy…)

Atgeirr


Have moved the * - when it comes to return type I am bit more uncertain. My rationale is that when we are closely mimicking an existing standard function - i.e. memcmp() in this case, we should follow the return value convention of that function.

Alternatively the function could be completely renamed to e.g. double_ptr_equal()and then return bool?

—
Reply to this email directly or view it on GitHub<https://github.com/OPM/opm-common/pull/74/files#r52300950>.